### ISSUE TITLE
Better Tree Shaking Support

### ISSUE THREAD

**Issue by elInfidel** [swagger-api/swagger-js#1970]
Hey,

I've noticed two libraries, esprima and js-yaml which are included into my build regardless of the fact I don't need them. These packages appear to be pulled in by swagger-client based on other issues. The solution currently is to modify web pack configuration. I use a non-ejected version of CRA and would prefer to see if I can make a contribution which improves this situation.

If I were to make a pull request, what would be involved to achieve a situation where tree shaking is better supported.

#1903 #960

**Comment by char0n** [comment]
Hi @elInfidel,

Sure git it shot. You can find more information about how current build is architected in this ticket https://github.com/swagger-api/swagger-js/issues/1542.

For your particular use-case - getting rid of esprima and js-yaml in non-ejected CRA is to use [craco](https://github.com/gsoft-inc/craco) and modify the webpack config and not bundle them in the way as I've suggested in #1903.

**Comment by wparad** [comment]
Latest version of `js-yaml` no longer requires `esprima`, swagger-client should just upgrade it's usage of the library.

**Comment by char0n** [comment]
@wparad yup, I've already created a [PR](https://github.com/swagger-api/swagger-js/pull/1867), we need to handle this update on swagger-ui in parallel too before I can merge this PR.

**Comment by elInfidel** [comment]
esprima is the main pain point for me personally. Excited to see it's no longer needed.

**Comment by char0n** [comment]
With [SwaggerUI@4.0.0](https://www.npmjs.com/package/swagger-ui/v/4.0.0-beta.1) and SwaggerEditor@4.0.0 pre-releases we have great opportunity to update [js-yaml](https://www.npmjs.com/package/js-yaml) to latest version on swagger-client, SwaggerUI and SwaggerEditor respectively.

Naturally this change needs to come in here (in swagger-client) first. Here is the **plan**:

1. Update js-yaml on swagger-client and release a patch version
2. Update js-yaml on SwaggerUI 3.x branch (current master)
3. Update js-yaml on SwaggerUI 4.x branch
4. Update js-yaml on SwaggerEditor 3.x branch (current master)
5. Update js-yaml on SwaggerEditor 4.x branch

There is an implication for older releases of SwaggerUI 3.x and SwaggerEditor 3.x though. If they will be reinstalled, they will pull newest version of swagger-client with updated js-yaml library. This can cause problems as there will be two versions of js-yaml installed within the dependency tree. The remedy for this is simple - recommend installing newest version of SwaggerUI/SwaggerEditor.

**Comment by char0n** [comment]
Closed by https://github.com/swagger-api/swagger-js/releases/tag/v3.15.0
